### ISSUE TITLE
By default NuGetForUnity downloads dozens of unneeded "sytem" packages, causing duplication compilation errors for NSubstitute.

### ISSUE THREAD

**Issue by spiralsam** [GlitchEnzo/NuGetForUnity#330]
## Description
If you download in a fresh Unity Project + NuGetForUnity then try to install NSubstitute nuget (versions below), then dozens of packages are installed with names like "System.Dynamic.Runtime.4.30" and "System.IO.4.3.0" and "runtime.debian.8-x64.runtime.native.System.Security.Cryptogrophay.OpenSsl.4.3.0". On and on.

NSubstitute should really just have Castle.Core.4.4.0 and System.Threading.Tasks.Extensions.4.3.0. In fact, after NuGetForUnity downloads all these extra packages, NuGet->Show Dependecy Tree does show only these. However in Nuget->Manage  Nuget Packages, all these extra packages are listed.

These extra unneeded packages then permanently cause the Project Build to fail building with error:
error CS1703: Multiple assemblies with equivalent identity have been imported: 

Then goes on to list various DLLs conflicted from the various .\Assets\Packages subdirectories.

## Versions
NSubstitute 4.2.1
NuGetForUnity 2.0.0
Unity 2020.2.0a10.1205 . Also tried this with Unity 2019.3.13.f1
Windows 10x64
All "options" for Unity/NuGetForUnity used here are essentially at default after creating new Unity Project.

**Comment by jwittner** [comment]
Curious what you're targetting for the API Compatibility Level?

**Comment by DanStevens** [comment]
I'm getting the same issue after installing the *NSubstitute* package, version 4.2.2. It appears to have installed a whole list of packages mostly in the `System` namespace and I'm getting compiler errors when I try to tun my test cases:

```
Microsoft (R) Visual C# Compiler version 2.9.1.65535 (9d34608e)
Copyright (C) Microsoft Corporation. All rights reserved.

error CS1703: Multiple assemblies with equivalent identity have been imported: 'C:\Users\Dan\Unity Projects\Inquisitor Combat System\Assets\Packages\System.Dynamic.Runtime.4.3.0\lib\netstandard1.3\System.Dynamic.Runtime.dll' and 'C:\Program Files\Unity\Hub\Editor\2019.4.0f1\Editor\Data\NetStandard\compat\2.0.0\shims\netstandard\System.Dynamic.Runtime.dll'. Remove one of the duplicate references.
error CS1703: Multiple assemblies with equivalent identity have been imported: 'C:\Users\Dan\Unity Projects\Inquisitor Combat System\Assets\Packages\System.ComponentModel.4.3.0\lib\netstandard1.3\System.ComponentModel.dll' and 'C:\Program Files\Unity\Hub\Editor\2019.4.0f1\Editor\Data\NetStandard\compat\2.0.0\shims\netstandard\System.ComponentModel.dll'. Remove one of the duplicate references.
error CS1703: Multiple assemblies with equivalent identity have been imported: 'C:\Users\Dan\Unity Projects\Inquisitor Combat System\Assets\Packages\System.Xml.XDocument.4.3.0\lib\netstandard1.3\System.Xml.XDocument.dll' and 'C:\Program Files\Unity\Hub\Editor\2019.4.0f1\Editor\Data\NetStandard\compat\2.0.0\shims\netstandard\System.Xml.XDocument.dll'. Remove one of the duplicate references.
error CS1703: Multiple assemblies with equivalent identity have been imported: 'C:\Users\Dan\Unity Projects\Inquisitor Combat System\Assets\Packages\System.Runtime.Numerics.4.3.0\lib\netstandard1.3\System.Runtime.Numerics.dll' and 'C:\Program Files\Unity\Hub\Editor\2019.4.0f1\Editor\Data\NetStandard\compat\2.0.0\shims\netstandard\System.Runtime.Numerics.dll'. Remove one of the duplicate references.
error CS1703: Multiple assemblies with equivalent identity have been imported: 'C:\Users\Dan\Unity Projects\Inquisitor Combat System\Assets\Packages\System.ObjectModel.4.3.0\lib\netstandard1.3\System.ObjectModel.dll' and 'C:\Program Files\Unity\Hub\Editor\2019.4.0f1\Editor\Data\NetStandard\compat\2.0.0\shims\netstandard\System.ObjectModel.dll'. Remove one of the duplicate references.
error CS1703: Multiple assemblies with equivalent identity have been imported: 'C:\Users\Dan\Unity Projects\Inquisitor Combat System\Assets\Packages\System.Collections.Concurrent.4.3.0\lib\netstandard1.3\System.Collections.Concurrent.dll' and 'C:\Program Files\Unity\Hub\Editor\2019.4.0f1\Editor\Data\NetStandard\compat\2.0.0\shims\netstandard\System.Collections.Concurrent.dll'. Remove one of the duplicate references.
error CS1703: Multiple assemblies with equivalent identity have been imported: 'C:\Users\Dan\Unity Projects\Inquisitor Combat System\Assets\Packages\System.Threading.4.3.0\lib\netstandard1.3\System.Threading.dll' and 'C:\Program Files\Unity\Hub\Editor\2019.4.0f1\Editor\Data\NetStandard\compat\2.0.0\shims\netstandard\System.Threading.dll'. Remove one of the duplicate references.
```
The *API Compatibility Level* setting (in <kbd>Edit</kbd> > <kbd>Project Settings</kbd> > <kbd>Player</kbd>) is set to '.Net Standard 2.0', so perhaps NuGetForUnity is installing .NET 4 versions of the system DLLs, since I expect it was developed for a new version of .NET.

I uninstalled the *NSubstitute* package and manually deleted all the imported packages and then removed NuGetForUnity so I could try again. This time I changed the *API Compatibility Level* setting to '.NET 4.x', reimported NuGetForUnity and installed the *NSubstitute* package. This time the only dependencies that got installed were *System.Threading.Tasks.Extensions.4.3.0* and *Castle.Core.4.4.0".

I understand why switching 'API Compatability Level' to '.NET 4.x' was needed if a .NET 4 NuGet package is installed, but I think there's room for improving the user experience. I think the user needs to be informed that the NuGet package depends on .NET 4.x. Additional features I would suggest, if possible, a filter to show only packages/versions that support .NET 2.0 or .NET 4.x and ability to install previous versions of packages.

**Comment by MaxwellDexter** [comment]
I was installing the SQLKata package and encountered this issue as well, swapping the API Compatibility Level to 4.x fixed the issue for me. To reiterate Dan's point, it'd be great if the asset told me about this and I didn't have to go through GitHub issues to fix it.

**Comment by jwittner** [comment]
Note: this may be resolved by #357 .

**Comment by KonH** [comment]
Looks like it will be resolved for both NSubstitute and SQLKata by #357 (tested on Unity 2019.4.0f1, .NET Standard 2.0 subset, MacOS)
